core/src/main/java/org/apache/iceberg/rest/HTTPClientFactory.java

```diff
@@ -0,0 +1,49 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iceberg.rest;
+
+import java.util.Map;
+import org.apache.iceberg.CatalogProperties;
+import org.apache.iceberg.relocated.com.google.common.base.Preconditions;
+
+/**
+ * Takes in the full configuration for the {@link RESTCatalog}, which should already have
+ * called the server's initial configuration route.
+ * Using the merged configuration, an instance of {@link RESTClient} is obtained that can be used with the
+ * RESTCatalog.
+ */
+public class HTTPClientFactory implements RESTClientFactory {
+
+  @Override
+  public RESTClient apply(Map<String, String> properties) {
+    Preconditions.checkArgument(properties != null, "Invalid initial configuration: null");
+    Preconditions.checkArgument(properties.containsKey(CatalogProperties.URI), "REST Catalog server URI is required");
+    Preconditions.checkArgument(properties.containsKey(RESTCatalogProperties.AUTH_TOKEN),
+        "Invalid http client configuration. Missing authentication token");
```

Comment:
Is this required? I think we should use it if it's there, but we don't need to force other people into using an auth token.

Reply:
I was wondering the same thing. I had originally had it as optional.

I'll change it so that it is only applied if the user provides it.

Reply:
Updated to use the auth token only if present.